### PR TITLE
182036856 vote accounts is active

### PR DESCRIPTION
**Pull request by maciejkocylapc**
#### What's this PR do?
- add is_active flag for vote_accounts and set it in VoteAccountDetailsService

#### How should this be manually tested?
- run tests
- run migrations
- run gather_vote_account_details_daemon.rb
- check if there are accounts with is_active flag set to false

#### What are the relevant tickets?
- [URL to the PivotalTracker ticket](https://www.pivotaltracker.com/story/show/182036856)
- https://www.pivotaltracker.com/story/show/182072088

#### Task completed checklist:
- [] Is there appropriate test coverage?
- [] Did I take into consideration possible security vulnerabilities?
- [] Are the controllers secure?
- [] Was every important matter documented?

#### Things to watch out for
- [Y] Does this PR have any migrations?
- [N] Does this PR add new dependencies?

### CODE REVIEW

**Review by radek-polcode — CHANGES_REQUESTED** [review]

File: app/services/gatherers/vote_account_details_service.rb [hunk 1]
```diff
@@ -9,25 +9,39 @@ class VoteAccountDetailsService
...
+                 .each do |vacc|
+        @range_start = vacc.id
+
         vote_account_details = get_vote_account_details(vacc.account)
-        unless vote_account_details.blank?
-          vacc.update(
+        if vote_account_details.blank?
+          vacc.set_inactive
+        else
+          if vacc.update(
```
Comment (radek-polcode) [inline app/services/gatherers/vote_account_details_service.rb:26]: You should check here if the values from solana are different than what we already have nad based on that update or not the record

File: app/models/vote_account.rb [hunk 2]
```diff
@@ -37,6 +38,12 @@ class VoteAccount < ApplicationRecord
   before_save :set_network
 
   scope :for_api, -> { select(FIELDS_FOR_API).order(updated_at: :asc) }
+  scope :active, -> { where(is_active: true) }
+
+  def set_inactive
+    self.is_active = false
+    self.save(touch: false)
```
[the remark this answers is not in the document]
Comment (maciejkocylapc) [reply app/models/vote_account.rb:45]: we never want to change timestamps if is_active is changed to false, but I've changed the method name to touch_inactive instead.

File: app/services/gatherers/vote_account_details_service.rb [hunk 3]
```diff
@@ -6,6 +6,11 @@ module Gatherers
   class VoteAccountDetailsService
     include SolanaLogic
 
+    SCORE_OPTIONS = {
```
Comment (radek-polcode) [inline app/services/gatherers/vote_account_details_service.rb:9]: I'd put that to the model and freeze that.

**Review by radek-polcode — CHANGES_REQUESTED** [review]

File: app/services/gatherers/vote_account_details_service.rb [hunk 4]
```diff
@@ -36,6 +45,11 @@ def call
 
     private
 
+    def should_omit_update?(vacc, vote_account_details)
+      vacc.validator_identity == vote_account_details["validatorIdentity"] && \
+      vacc.authorized_withdrawer == vote_account_details["authorizedWithdrawer"]
```
Comment (radek-polcode) [inline app/services/gatherers/vote_account_details_service.rb:50]: Does this make sense to check `is_active` as well?

File: test/services/gatherers/vote_account_details_service_test.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      validatorIdentity: @validator_identity,
+      authorizedWithdrawer: @authorized_withdrawer
+    }.to_json
+
+    @val = create(:validator, network: @network)
+    @score = create(:validator_score_v1, network: @network, validator: @val)
+  end
+
+  teardown do
+    VoteAccount.delete_all
```
Comment (radek-polcode) [inline test/services/gatherers/vote_account_details_service_test.rb:22]: Is that needed?

File: test/services/gatherers/vote_account_details_service_test.rb [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+      assert_equal 0, @score.reload.authorized_withdrawer_score
+    end
+  end
+
+  test "VoteAccountDetailsService sets correct score" do
+    va1 = create(
+      :vote_account,
+      validator: @val,
+      network: @network,
+      account: "acc1",
```
Comment (radek-polcode) [inline test/services/gatherers/vote_account_details_service_test.rb:46]: I think this could be initialized in setup so you can use it in all tests, and two attributes below you can update here in test.

**Review by radek-polcode — CHANGES_REQUESTED** [review]

**Review by radek-polcode — CHANGES_REQUESTED** [review]
I think that we also need to test case when solana response is empty, if should_omit is true and if record update fails.

File: test/services/gatherers/vote_account_details_service_test.rb [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+      Gatherers::VoteAccountDetailsService.new(network: @network, config_urls: [@testnet_url]).call
+
+      assert_equal @va1.reload.validator_identity, @validator_identity
+      assert_equal @va1.reload.authorized_withdrawer, @authorized_withdrawer
+      assert_equal 0, @score.reload.authorized_withdrawer_score
+    end
+  end
+
+  test "call sets negative score correctly" do
+    @json_response = {
```
Comment (radek-polcode) [inline test/services/gatherers/vote_account_details_service_test.rb:39]: Does this need to be instance variable? Similar in other tests.

**Review by radek-polcode — CHANGES_REQUESTED** [review]

**Review by kbiala — APPROVED** [review]
